src/pools/fixedRate/FixedRateExchange.ts

```diff
@@ -362,13 +362,17 @@ export class FixedRateExchange {
     exchangeId: string,
     newRate: string
   ): Promise<TransactionReceipt> {
+    const exchange = await this.getExchange(exchangeId)
     const estGas = await this.estSetRate(
       address,
       exchangeId,
-      this.web3.utils.toWei(String(newRate))
+      await this.amountToUnits(exchange.baseToken, newRate.toString())
```

Comment:
this conversions should also take place in `estSetRate()` just pass the raw string values to the estimate method

Reply:
right, makes sense